pydantic_settings/sources.py

```diff
@@ -594,14 +594,17 @@ def __call__(self) -> dict[str, Any]:
         if self.secrets_dir is None:
             return secrets
 
-        self.secrets_path = Path(self.secrets_dir).expanduser()
+        secrets_dirs = [self.secrets_dir] if isinstance(self.secrets_dir, (str, os.PathLike)) else self.secrets_dir
+        self.secrets_paths = [Path(p).expanduser() for p in secrets_dirs]
 
-        if not self.secrets_path.exists():
-            warnings.warn(f'directory "{self.secrets_path}" does not exist')
-            return secrets
+        for path in self.secrets_paths:
+            if not path.exists():
+                warnings.warn(f'directory "{path}" does not exist')
+                return secrets
 
-        if not self.secrets_path.is_dir():
-            raise SettingsError(f'secrets_dir must reference a directory, not a {path_type_label(self.secrets_path)}')
+        for path in self.secrets_paths:
+            if not path.is_dir():
+                raise SettingsError(f'secrets_dir must reference a directory, not a {path_type_label(path)}')
```

Comment:
There is a test for this error that tests single dir. please add a test for multiple directories as well.